src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,202 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public MyLinkedList() {
+        this.size = 0;
+        this.head = null;
+        this.tail = null;
+    }
+
     @Override
     public void add(T value) {
+        addTail(value);
     }
 
     @Override
     public void add(T value, int index) {
+        int linkedListSize = size;
+        checkOnIndexOutOfBoundsException(index, false);
+
+        if (linkedListSize == 0) {
+            addHead(value);
+        } else if (linkedListSize == index) {
+            addTail(value);
+        } else {
+            Node<T> findedNode = findNode(index);
+            addBetween(value, findedNode);
```

Comment:
```suggestion
            addBefore(value, findedNode);
```

Reply:
I'm not agree with you, for me this method like add between, not before